instrumentation/apache-dubbo-2.7/library/build.gradle.kts

```diff
@@ -3,6 +3,12 @@ plugins {
 }
 
 dependencies {
+  compileOnly("com.google.auto.value:auto-value-annotations")
+  annotationProcessor("com.google.auto.value:auto-value")
+
+  // required for PeerServiceAttributesExtractor
+  implementation(project(":javaagent-instrumentation-api"))
```

Comment:
The library instrumentation here is a bit unusual, because the only way to inject a `Filter` is through SPI - so we have to use `GlobalOpenTelemetry`, `PeerServiceExtractor` et cetera in the library instrumentation; customization is just not possible.

Reply:
could we have a separate SPI class that we inject via the javaagent (which does some init and delegates to the library SPI class)?

Reply:
That was my first idea (since AWS SDK does this) - but unfortunately it's not possible to inject a `Filter` manually, there is no way to add it to Dubbo except for the SPI. So the users of the library instrumentation would have to create a delegate `Filter` implementation with correct `@Activate` annotation and add SPI file to make it work - that's a lot of work required just to make it work.

Reply:
sent a PR to your branch with a possible idea https://github.com/mateuszrzeszutek/opentelemetry-java-instrumentation/pull/29

Reply:
Thanks! I'll try that out

Reply:
Okay, done - I changed your version a bit and added those "standard" library instrumentation `DubboTracing`/`DubboTracingBuilder` classes.